src/conshdlr.jl

```diff
@@ -0,0 +1,125 @@
+# Current limitations:
+# - use fixed values for some properties
+#   - EAGERFREQ = 100
+#   - NEEDSCONS = TRUE
+#   - SEPAPRIORITY = 0
+#   - SEPAFREQ = -1
+#   - DELAYSEPA = FALSE
+#   - PROPFREQ = -1
+#   - DELAYPROP = -1
+#   - PROP_TIMING = SCIP_PROPTIMING_BEFORELP
+#   - PRESOLTIMING = SCIP_PRESOLTIMING_MEDIUM
+#   - MAXPREROUNDS = -1
+# - don't support optional methods: CONSHLDRCOPY, CONSFREE, CONSINIT, CONSEXIT,
+#   CONSINITPRE, CONSEXITPRE, CONSINITSOL, CONSEXITSOL, CONSDELETE, CONSTRANS,
+#   CONSINITLP, CONSSEPALP, CONSSEPASOL, CONSENFORELAX, CONSPROP, CONSPRESOL,
+#   CONSRESPROP, CONSACTIVE, CONSDEACTIVE, CONSENABLE, CONSDISABLE, CONSDELVARS,
+#   CONSPRINT, CONSCOPY, CONSPARSE, CONSGETVARS, CONSGETNVARS, CONSGETDIVEBDCHGS
+# - don't support linear or nonlinear constraint upgrading
```

Comment:
What is constraint upgrading?

Reply:
Constraint upgrading means detecting special structure, and then converting a given constraint to a constraint of a different handler. For example, a user might create a linear constraint, but then SCIP detects that it is actually a set-cover constraint, so it will create a constraint of that type in order to use specialized methods for cuts, presolve etc. Similarly, a user might submit a constraint with a nonlinear, but then SCIP detects that it can be represented as a quadratic constraint handler and upgrades to that. Not sure whether the original constraints remain in the (transformed) problem or not.